enhancements/olm/single-ownnamespace-enhancement-proposal.md

```diff
@@ -0,0 +1,529 @@
+- **Role/RoleBinding**: `permissions` entries in the CSV are created as `Role` and `RoleBinding` resources in the watch namespace(s). For AllNamespaces mode, these are instead created as `ClusterRole` and `ClusterRoleBinding` resources
+- **Operator Configuration**: `olm.targetNamespaces` annotation gets set in the operator deployment's pod template, instructing the operator how to configure itself for the target namespace scope
+
+`Note`: With watch namespace as a configuration value that can (at times, must) be provided for installation, the user input will be validated based on the bundle's supported install modes:
+
+| AllNamespaces | SingleNamespace | OwnNamespace | WatchNamespace Configuration                                     |
+|---------------|-----------------|--------------|------------------------------------------------------------------|
+| -             | -               | -            | undefined/error (no supported install modes)                     |
+| -             | -               | ✓            | no configuration                                                 |
+| -             | ✓               | -            | required (must not be install namespace)                         |
+| -             | ✓               | ✓            | optional (default: install namespace)                            | 
+| ✓             | -               | -            | no configuration                                                 | 
+| ✓             | -               | ✓            | optional. If set, must be install namespace (default: unset)     | 
+| ✓             | ✓               | -            | optional. If set, must NOT be install namespace (default: unset) | 
+| ✓             | ✓               | ✓            | optional (default: unset)                                        | 
+
+Add support for namespace-scoped operation by ensuring:
+1. Bundle validation confirms the operator supports the requested install mode
+2. RBAC resources are properly scoped to the watch namespace for Single/OwnNamespace modes
+3. Operator deployment environment variables are configured for the specified watch namespace
+4. Installation fails gracefully with clear error messages when bundles don't support the requested mode
+5. Usage of reasonable default install mode when possible: 
+  - AllNamespaces by default (when available)
+  - OwnNamespace by default (when no AllNamespaces and available)
+
+Ensure parity with OLMv0 behavior by:
+1. Generating identical RBAC and deployment configurations for equivalent install modes
+2. Maintaining the same namespace isolation boundaries
+3. Preserving operator functionality across install mode transitions
+
+### Workflow Description
+
+#### Administrator Workflow
+
+1. **Enable Feature Gate**: Administrator enables the `SingleOwnNamespaceInstallSupport` feature gate on the operator-controller deployment
```

Comment:
We should either remove this (as it will be enabled in GA), or call out what "while in TP" and use the downstream feature gate name `NewOLMOwnSingleNamespace`

Reply:
That's a good point, removing this altogether so that the documentation in other places can take care of listing if this step is needed or not.